Client/game_sa/CPedSA.cpp

```diff
@@ -854,7 +854,7 @@ void CPedSA::SetBleeding(bool bBleeding)
 bool CPedSA::SetOnFire(bool onFire)
 {
     CPedSAInterface* pInterface = GetPedInterface();
-    if (onFire == !!pInterface->pFireOnPed)
+    if (onFire && pInterface->pFireOnPed)
```

Comment:
`setPedOnFire(ped, false)` doesn't return false if the ped wasn't on fire

Reply:
Keep the oroginal check and you don't need your `wasFire` variable

Reply:
I think the function should return true even if there was no fire but the task was removed

Reply:
Old code returns `false` here when:
* a ped is on fire, and you are trying to enable fire status.
* a ped is not on fire and you are trung to disable fire status.

Your check doesn't return false in the second case